File: Third_Year/Semester1/Mobile/Lab2(native)/app/src/main/java/com/example/expensewizard/model/Transaction.kt
package com.example.expensewizard.model

import android.util.Log
import java.util.Date
data class Transaction (
    var id : Int,
    var title: String,
    var type:Type,
    var category: String,
    var amount : Double,
    var timestamp : Date
) {
    companion object {
        var currentId = 0
    }

    constructor(title: String, type: Type, category: String, amount: Double, timestamp: Date) :
            this(currentId++,title,type,category,amount,timestamp)
    {
        Log.i("Model Transaction Class: ", "Current id is $currentId") }


}

File: Third_Year/Semester1/Mobile/Lab3(non-native)/non_native/lib/model/transaction.dart
import 'type.dart';

class Transaction {
  int id;
  String title;
  Type type;
  String category;
  double amount;
  DateTime timestamp;

  static int currentId = 0;

  Transaction({
    required this.id,
    required this.title,
    required this.type,
    required this.category,
    required this.amount,
    required this.timestamp,
});
  
  factory Transaction.createTransaction(
      String title,
      Type type,
      String category,
      double amount,
      DateTime timestamp,
      ) {
    return Transaction(id: currentId++, title: title, type: type, category: category, amount: amount, timestamp: timestamp);
  }

  static Transaction newTransaction(
      String title,
      Type type,
      String category,
      double amount,
      DateTime timestamp,
      ) {
    return Transaction(id: currentId++, title: title, type: type, category: category, amount: amount, timestamp: timestamp);

  }
}
